middlewares/auth.js

```diff
@@ -0,0 +1,32 @@
+const { User } = require("../service/Schemas/userSchema");
+const jwt = require("jsonwebtoken");
+
+require("dotenv").config();
+const { SECRET } = process.env;
+
+const auth = async (req, res, next) => {
+  const { authorization } = req.headers;
+  if (authorization) {
+    const token = authorization.split(" ")[1];
+
+    try {
+      const payload = jwt.verify(token, SECRET);
+      const { id } = payload;
+      const user = await User.findById({ _id: id });
+      if (!user || user.token !== token) {
+        return res.status(401).json({
+          message: "Not authorized",
+        });
+      } else {
+        req.user = user;
+        next();
+      }
```

Comment:
```suggestion
        req.user = user;
        next();
```
Tego else nie potrzebujesz bo masz return na gorze i ono zatrzyma wykonanie.